Sources/Sentry/SentryHub.m

```diff
@@ -382,22 +382,35 @@ - (SentryTracer *)startTransactionWithContext:(SentryTransactionContext *)transa
                         customSamplingContext:(NSDictionary<NSString *, id> *)customSamplingContext
                                 configuration:(SentryTracerConfiguration *)configuration
 {
-    SentrySamplingContext *samplingContext =
-        [[SentrySamplingContext alloc] initWithTransactionContext:transactionContext
-                                            customSamplingContext:customSamplingContext];
-
-    SentryTracesSamplerDecision *samplerDecision = [_tracesSampler sample:samplingContext];
-    transactionContext = [self transactionContext:transactionContext
-                                      withSampled:samplerDecision.decision];
-    transactionContext.sampleRate = samplerDecision.sampleRate;
+    SentryTracesSamplerDecision *samplerDecision;
 
+    // in order to sample launch profiles, we must have already computed the trace/profile sampling
+    // decisions, so that we don't run the profiler for a launch that it would eventually be decided
+    // to sample out, thus incurring unnecessary overhead. They are stored in a static reference in
+    // SentryTracer; access them here so that we can properly start the needed trace that will
+    // contain the launch profile.
 #if SENTRY_TARGET_PROFILING_SUPPORTED
-    SentryProfilesSamplerDecision *profilesSamplerDecision =
-        [_profilesSampler sample:samplingContext tracesSamplerDecision:samplerDecision];
+    SENTRY_LOG_DEBUG(@"Checking for a possible app launch sample decision...");
+    if (isTracingAppLaunch) {
```

Comment:
`m`: What about modifying the sampling context in the tracer to keep the app launch profiling logic in one class and leaving the logic here untouched? Then, we don't need to expose the `isTracingAppLaunch`.

Reply:
Just a note to myself mostly but I just realized I'd removed a piece of this logic that adds the profiles sampler decision to the configuration. Moving to the marker-file approach means I need to store the profile sample rate to a file so we can recover it because we need it to instantiate the SentryProfilesSamplerDecision in the case that a launch profile is running here.

Reply:
Now that this logic has moved to SentryLaunchProfiling, it's kind of weird to make that file responsible for _always_ modifying the sampling context, because this part of the hub isn't only dealing with launch profile transactions, but _all_ transactions. We'd have to expose `isTracingAppLaunch` to use it from `SentryTracer` as well

I moved what I could into another function in SentryLaunchProfiling, but wasn't able to fully eliminate the changes in SentryHub.startTransaction, although it is cleaner there now.